### PR TITLE
circuit: improve alloc entry log

### PR DESCRIPTION
**Pull request by tcoratger**
## Summary
What does this PR change?

I've tested the implementation results with the following run `RUST_LOG=DEBUG cargo run --example allocation_log --features debugging`

- Before

```shell
Building a simple circuit to demonstrate the allocation log...

Creating circuit: result = (a + b * c) * (a - b) / 2

DEBUG    🐛 [debug]: === Circuit Allocation Log ===
DEBUG    🐛 [debug]: Total allocations: 9

DEBUG    🐛 [debug]: === Allocation Log for scope 'main' ===
DEBUG    🐛 [debug]: Total allocations in scope: 9

DEBUG    🐛 [debug]: --- Public Inputs (3) ---
DEBUG    🐛 [debug]:   expr_1 (Public): input_a
DEBUG    🐛 [debug]:   expr_2 (Public): input_b
DEBUG    🐛 [debug]:   expr_3 (Public): input_c
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: --- Constants (1) ---
DEBUG    🐛 [debug]:   expr_4 (Const): 2
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: --- Additions (1) ---
DEBUG    🐛 [debug]:   expr_6 = expr_1 + expr_5: a_plus_bc
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: --- Subtractions (1) ---
DEBUG    🐛 [debug]:   expr_7 = expr_1 - expr_5: a_minus_bc
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: --- Multiplications (2) ---
DEBUG    🐛 [debug]:   expr_5 = expr_2 * expr_3: b_times_c
DEBUG    🐛 [debug]:   expr_8 = expr_6 * expr_7: sum_times_diff
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: --- Divisions (1) ---
DEBUG    🐛 [debug]:   expr_9 = expr_8 / expr_4: final_result
DEBUG    🐛 [debug]:
DEBUG    🐛 [debug]: === End Scope Log ===

DEBUG    🐛 [debug]: === End Allocation Log ===
```

- After

```shell
Building a simple circuit to demonstrate the allocation log...

Creating circuit: result = (a + b * c) * (a - b) / 2

DEBUG    circuit_allocation_log [ 111µs | 16.89% / 100.00% ] total: 9
DEBUG    ┝━ 🐛 [debug]: allocation summary | publics: 3 | constants: 1 | additions: 1 | subtractions: 1 | multiplications: 2 | divisions: 1 | horner_steps: 0 | non_primitive_ops: 0 | witness_hints: 0
DEBUG    ┕━ scope [ 92.5µs | 55.60% / 83.11% ] name: "main" | allocations: 9
DEBUG       ┝━ group [ 9.92µs | 8.91% ] name: "Public Inputs" | count: 3
DEBUG       │  ┝━ 🐛 [debug]:  | expr_id: 1 | alloc_type: Public | label: "input_a" | expression: expr_1 (Public)
DEBUG       │  ┝━ 🐛 [debug]:  | expr_id: 2 | alloc_type: Public | label: "input_b" | expression: expr_2 (Public)
DEBUG       │  ┕━ 🐛 [debug]:  | expr_id: 3 | alloc_type: Public | label: "input_c" | expression: expr_3 (Public)
DEBUG       ┝━ group [ 3.79µs | 3.41% ] name: "Constants" | count: 1
DEBUG       │  ┕━ 🐛 [debug]:  | expr_id: 4 | alloc_type: Const | label: "2" | expression: expr_4 (Const)
DEBUG       ┝━ group [ 3.83µs | 3.44% ] name: "Additions" | count: 1
DEBUG       │  ┕━ 🐛 [debug]:  | expr_id: 6 | alloc_type: Add | label: "a_plus_bc" | expression: expr_6 = expr_1 + expr_5
DEBUG       ┝━ group [ 3.50µs | 3.14% ] name: "Subtractions" | count: 1
DEBUG       │  ┕━ 🐛 [debug]:  | expr_id: 7 | alloc_type: Sub | label: "a_minus_bc" | expression: expr_7 = expr_1 - expr_5
DEBUG       ┝━ group [ 5.96µs | 5.35% ] name: "Multiplications" | count: 2
DEBUG       │  ┝━ 🐛 [debug]:  | expr_id: 5 | alloc_type: Mul | label: "b_times_c" | expression: expr_5 = expr_2 * expr_3
DEBUG       │  ┕━ 🐛 [debug]:  | expr_id: 8 | alloc_type: Mul | label: "sum_times_diff" | expression: expr_8 = expr_6 * expr_7
DEBUG       ┕━ group [ 3.62µs | 3.26% ] name: "Divisions" | count: 1
DEBUG          ┕━ 🐛 [debug]:  | expr_id: 9 | alloc_type: Div | label: "final_result" | expression: expr_9 = expr_8 / expr_4
```

Fixes #<issue-id>

## Why
What problem does this solve? Why is this approach correct?

## Changes
- [ ] Key change 1
- [ ] Key change 2

## Acceptance criteria
- [ ] <explicit condition that proves "done">
- [ ] <explicit condition that proves "done">

## Tests
- [ ] Unit tests added/updated
- [ ] Existing tests pass
- [ ] Edge cases considered (briefly note any)

## Benchmarks / Performance (if relevant)
- [ ] Before/after numbers included
- [ ] Benchmark command or link to CI perf output
- [ ] No unexpected regressions in CPU / memory / proof size / compile time

## Docs (if relevant)
- [ ] Rustdoc updated
- [ ] Book/README updated
- [ ] Examples updated

## Risk
What could break?

## Review notes
Where should reviewers focus on? Any tricky parts / follow-ups?

### CODE REVIEW

File: circuit/src/alloc_entry.rs [hunk 1]
```diff
@@ -25,304 +39,525 @@ pub enum AllocationType {
...
+            Self::Public => "Public Inputs",
+            Self::Const => "Constants",
+            Self::Add => "Additions",
+            Self::Sub => "Subtractions",
+            Self::Mul => "Multiplications",
+            Self::Div => "Divisions",
+            Self::HornerAcc => "Horner Accumulator Steps",
+            Self::NonPrimitiveOp(_) => "Non-Primitive Operations",
+            Self::WitnessHint => "Witness Hints",
+        }
```
Comment (Nashtare) [inline circuit/src/alloc_entry.rs]: why isn't this using `GROUP_ORDER`?

File: circuit/src/alloc_entry.rs [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+    "Public Inputs",
+    "Constants",
+    "Additions",
+    "Subtractions",
+    "Multiplications",
+    "Divisions",
+    "Horner Accumulator Steps",
+    "Non-Primitive Operations",
+    "Witness Hints",
+];
```
Comment (Nashtare) [inline circuit/src/alloc_entry.rs]: is this even needed? We could match `AllocationType` enum variants instead no?

**Review by Nashtare — COMMENTED** [review]
Overall this LGTM, I think this may be a better approach for feeding to LLMs / debugging scripts as actual circuits end up being too complex for human eyes anyway, but I left 2 comments inline, I think we can simplify the overhead

File: circuit/src/alloc_entry.rs (hunk 2 above)
Comment (tcoratger) [reply circuit/src/alloc_entry.rs]: Just proposed something, let me know if this is ok for you.

**Comment by tcoratger** [comment]
> Overall this LGTM, I think this may be a better approach for feeding to LLMs / debugging scripts as actual circuits end up being too complex for human eyes anyway, but I left 2 comments inline, I think we can simplify the overhead

Thanks a lot, just fixed all the comments :)
